### PR TITLE
Integration of the new pod recovery monitoring strategy implemented in krkn-lib

### PR DESCRIPTION
**Pull request by tsebastiani**

### CODE REVIEW

**Comment by paigerube14** [comment]
When I ran the etcd pod scenairo with the default pod recovery time in the config, I got the below error. It wanst exactly clear what was happening.  Once I increased the pod recovery timeout this functionality worked as expected

```
 % python run_kraken.py
 _              _              
| | ___ __ __ _| | _____ _ __  
| |/ / '__/ _` | |/ / _ \ '_ \ 
|   <| | | (_| |   <  __/ | | |
|_|\_\_|  \__,_|_|\_\___|_| |_|
                               

2024-04-15 15:41:19,682 [INFO] Starting kraken
2024-04-15 15:41:19,700 [INFO] Initializing client to talk to the Kubernetes cluster
2024-04-15 15:41:19,700 [INFO] Generated a uuid for the run: d6492275-fb64-41ae-bba0-1a10905e66c9
2024-04-15 15:41:23,352 [INFO] Publishing kraken status at http://0.0.0.0:8081
2024-04-15 15:41:28,354 [INFO] Starting http server at http://0.0.0.0:8081

2024-04-15 15:41:28,355 [INFO] Fetching cluster info
2024-04-15 15:41:28,355 [INFO] Cluster version CRD not detected, skipping
2024-04-15 15:41:28,356 [INFO] Server URL: https://api.**qe.gcp.devcluster.openshift.com:6443
2024-04-15 15:41:28,356 [INFO] Daemon mode not enabled, will run through 1 iterations

2024-04-15 15:41:28,356 [INFO] Executing scenarios for iteration 0
2024-04-15 15:41:28,357 [INFO] connection set up
127.0.0.1 - - [15/Apr/2024 15:41:28] "GET / HTTP/1.1" 200 -
2024-04-15 15:41:28,359 [INFO] response RUN
2024-04-15 15:41:28,362 [INFO] scenario scenarios/openshift/etcd.yml
2024-04-15 15:41:30,488 [INFO] {
        "output_id": "success",
        "output_data": {
                "pods": {
                        "1713210089440470000": {
                                "namespace": "openshift-etcd",
                                "name": "etcd-prubenda-***-master-1"
                        }
                }
        }
}

2024-04-15 15:41:30,488 [INFO] waiting 30 seconds for pods to recover...
2024-04-15 15:41:59,107 [ERROR] scenario exception: unrecovered pods: pod: etcd-***-master-1 namespace:openshift-etcd
2024-04-15 15:41:59,108 [ERROR] scenario: scenarios/openshift/etcd.yml failed with exception: <class 'Exception'> file: /Users/prubenda/PycharmProjects/pplRepos/tsebastiani/krkn/kraken/plugins/__init__.py line: 277
2024-04-15 15:41:59,108 [INFO] 
2024-04-15 15:42:07,358 [INFO] Chaos data:
{
    "telemetry": {
        "scenarios": [
            {
                "start_timestamp": 0,
                "end_timestamp": 0,
                "scenario": "scenarios/openshift/etcd.yml",
                "exit_status": 0,
                "parameters_base64": "",
                "parameters": [
                    {
                        "config": {
                            "label_selector": "k8s-app=etcd",
                            "namespace_pattern": "^openshift-etcd$"
                        },
                        "id": "kill-pods"
                    }
                ],
                "affected_pods": {
                    "recovered": [],
                    "unrecovered": [],
                    "error": "pod: etcd-***-master-1 namespace:openshift-etcd"
                }
            }
        ],
```

File: config/config.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ kraken:
     signal_state: RUN                                      # Will wait for the RUN signal when set to PAUSE before running the scenarios, refer docs/signal.md for more details
     signal_address: 0.0.0.0                                # Signal listening address
     port: 8081                                             # Signal port
+    pod_recovery_time: 30                                  # pod recovery time after kill pod scenario.
```
Comment (chaitanyaenr) [inline config/config.yaml:9]: Pod scenario config might be a better place for setting this I guess. Thoughts?

Comment (paigerube14) [reply config/config.yaml:9]: +1, we might want to set different times based on different scenarios

**Comment by paigerube14** [comment]
Think we need to add the krkn_pod_recovery_time to the validation for the pod scenario, getting the below error: 
```
2024-04-19 15:44:00,765 [INFO] waiting 30 seconds for pod recovery, pod label selector: k8s-app=etcd namespace pattern: ^openshift-etcd$
2024-04-19 15:44:00,768 [ERROR] scenario exception: Validation failed for 'KillPodConfig': Invalid parameter 'krkn_pod_recovery_time', expected one of: namespace_pattern, name_pattern, kill, label_selector, kubeconfig, timeout, backoff
2024-04-19 15:44:00,769 [ERROR] scenario: scenarios/openshift/etcd.yml failed with exception: <class 'arcaflow_plugin_sdk.schema.ConstraintException'> file: /Users/prubenda/PycharmProjects/pplRepos/tsebastiani/krkn/kraken/plugins/__init__.py line: 271
2024-04-19 15:44:00,769 [INFO] 
```

**Comment by tsebastiani** [comment]
> Think we need to add the krkn_pod_recovery_time to the validation for the pod scenario, getting the below error:
> 
> ```
> 2024-04-19 15:44:00,765 [INFO] waiting 30 seconds for pod recovery, pod label selector: k8s-app=etcd namespace pattern: ^openshift-etcd$
> 2024-04-19 15:44:00,768 [ERROR] scenario exception: Validation failed for 'KillPodConfig': Invalid parameter 'krkn_pod_recovery_time', expected one of: namespace_pattern, name_pattern, kill, label_selector, kubeconfig, timeout, backoff
> 2024-04-19 15:44:00,769 [ERROR] scenario: scenarios/openshift/etcd.yml failed with exception: <class 'arcaflow_plugin_sdk.schema.ConstraintException'> file: /Users/prubenda/PycharmProjects/pplRepos/tsebastiani/krkn/kraken/plugins/__init__.py line: 271
> 2024-04-19 15:44:00,769 [INFO] 
> ```

I've already added, did you pull the latest version from the repo? `pip install --force-reinstall git+https://github.com/krkn-chaos/arcaflow-plugin-kill-pod.git
`

File: scenarios/openshift/etcd.yml [hunk 2]
```diff
@@ -3,8 +3,4 @@
   config:
     namespace_pattern: ^openshift-etcd$
     label_selector: k8s-app=etcd
-- id: wait-for-pods
-  config:
-    namespace_pattern: ^openshift-etcd$
-    label_selector: k8s-app=etcd
-    count: 3
+    krkn_pod_recovery_time: 30
```
Comment (paigerube14) [inline scenarios/openshift/etcd.yml:6]: Tullio, are we able to increase the recovery times in the scenarios to 120 seconds to start? The etcd pod scenario I ran took 97 seconds:         
Also, should we document somewhere what each of the times below are? Is readiness the pod is created and then thats the time it takes to become ready? 
```
"total_recovery_time": 107.39545583724976,
"pod_readiness_time": 9.480031967163086,
"pod_rescheduling_time": 97.91542387008667
```

**Review by paigerube14 — APPROVED** [review]
/lgtm

**Pull request merged**